### PR TITLE
[CPU][ARM] Set `arm64-v8.2-a` as a default architecture

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 1]
```diff
@@ -32,12 +32,8 @@ if(ARM)
...
                                 ${OV_CPU_ARM_TARGET_GENERIC_ARCHS})
 elseif(AARCH64)
-    if(APPLE)
-        # Apple M1 / M2 is assumed
-        set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8.2-a)
-    else()
-        set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8a)
-    endif()
+    # Apple M1 / M2 or Raspberry 5 is assumed
+    set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8.2-a)
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/CMakeLists.txt:36]: Future ARM Compute Library releases will have an ability to compile universal library which will support fp32, fp16, SVE, SVE2 kernels in the same library.
Based on this, we can compile a single plugin for all our ARM64 targets and don't have to change base arch.

See https://github.com/ARM-software/ComputeLibrary/issues/1053#issuecomment-1845611959

**Review by ilya-lavrenov — APPROVED** [review]
Default builds in Jenkins are switched to arm64-v8a.
GHA builds will use these newer arch and test CPU plugin on Ampere server

**Comment by ilya-lavrenov** [comment]
Looks like you have to fix some compilation issues:
```
2023-12-12T08:16:04.6763739Z In file included from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/memory_desc/cpu_blocked_memory_desc.h:8,
2023-12-12T08:16:04.6765383Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/memory_desc/dnnl_memory_desc.h:7,
2023-12-12T08:16:04.6766699Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/cpu_memory.h:14,
2023-12-12T08:16:04.6767850Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/node.h:15,
2023-12-12T08:16:04.6769023Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.h:8,
2023-12-12T08:16:04.6770274Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:5:
2023-12-12T08:16:04.6772831Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h: In instantiation of 'constexpr bool ov::intel_cpu::one_of(T, P, Args ...) [with T = long unsigned int; P = int; Args = {int}]':
2023-12-12T08:16:04.6775286Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:211:54:   required from here
2023-12-12T08:16:04.6833382Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:32:16: error: comparison of integer expressions of different signedness: 'long unsigned int' and 'int' [-Werror=sign-compare]
2023-12-12T08:16:04.6835345Z    32 |     return val == item || one_of(val, item_others...);
2023-12-12T08:16:04.6835998Z       |            ~~~~^~~~~~~
2023-12-12T08:16:04.6837873Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h: In instantiation of 'constexpr bool ov::intel_cpu::one_of(T, P) [with T = long unsigned int; P = int]':
2023-12-12T08:16:04.6841351Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:32:33:   required from 'constexpr bool ov::intel_cpu::one_of(T, P, Args ...) [with T = long unsigned int; P = int; Args = {int}]'
2023-12-12T08:16:04.6843639Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:211:54:   required from here
2023-12-12T08:16:04.6846101Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:28:51: error: comparison of integer expressions of different signedness: 'long unsigned int' and 'int' [-Werror=sign-compare]
2023-12-12T08:16:04.6848048Z    28 | constexpr bool one_of(T val, P item) { return val == item; }
2023-12-12T08:16:04.6848825Z       |                                               ~~~~^~~~~~~
2023-12-12T08:16:04.6849481Z cc1plus: all warnings being treated as errors
```

**Comment by alvoron** [comment]
> Looks like you have to fix some compilation issues:
> 
> ```
> 2023-12-12T08:16:04.6763739Z In file included from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/memory_desc/cpu_blocked_memory_desc.h:8,
> 2023-12-12T08:16:04.6765383Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/memory_desc/dnnl_memory_desc.h:7,
> 2023-12-12T08:16:04.6766699Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/cpu_memory.h:14,
> 2023-12-12T08:16:04.6767850Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/node.h:15,
> 2023-12-12T08:16:04.6769023Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.h:8,
> 2023-12-12T08:16:04.6770274Z                  from /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:5:
> 2023-12-12T08:16:04.6772831Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h: In instantiation of 'constexpr bool ov::intel_cpu::one_of(T, P, Args ...) [with T = long unsigned int; P = int; Args = {int}]':
> 2023-12-12T08:16:04.6775286Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:211:54:   required from here
> 2023-12-12T08:16:04.6833382Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:32:16: error: comparison of integer expressions of different signedness: 'long unsigned int' and 'int' [-Werror=sign-compare]
> 2023-12-12T08:16:04.6835345Z    32 |     return val == item || one_of(val, item_others...);
> 2023-12-12T08:16:04.6835998Z       |            ~~~~^~~~~~~
> 2023-12-12T08:16:04.6837873Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h: In instantiation of 'constexpr bool ov::intel_cpu::one_of(T, P) [with T = long unsigned int; P = int]':
> 2023-12-12T08:16:04.6841351Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:32:33:   required from 'constexpr bool ov::intel_cpu::one_of(T, P, Args ...) [with T = long unsigned int; P = int; Args = {int}]'
> 2023-12-12T08:16:04.6843639Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/nodes/reorder.cpp:211:54:   required from here
> 2023-12-12T08:16:04.6846101Z /__w/openvino/openvino/openvino/src/plugins/intel_cpu/src/utils/general_utils.h:28:51: error: comparison of integer expressions of different signedness: 'long unsigned int' and 'int' [-Werror=sign-compare]
> 2023-12-12T08:16:04.6848048Z    28 | constexpr bool one_of(T val, P item) { return val == item; }
> 2023-12-12T08:16:04.6848825Z       |                                               ~~~~^~~~~~~
> 2023-12-12T08:16:04.6849481Z cc1plus: all warnings being treated as errors
> ```

Comparison issue is fixed. Now some C++ unit tests and Python tests are failed. Could you please confirm that GHA arm pipelines are green on master? It seems to me that unit tests failures are not related to the proposed changes.

**Comment by ilya-lavrenov** [comment]
> Comparison issue is fixed. Now some C++ unit tests and Python tests are failed. Could you please confirm that GHA arm pipelines are green on master? It seems to me that unit tests failures are not related to the proposed changes.

Yes, Linux ARM64 is green on master https://github.com/openvinotoolkit/openvino/actions/runs/7187462163/job/19575072517

Your changes lead to new kernels selected in ARM Compute Library, which can definitely lead to some accuracy issues.

**Review by dmitry-gorokhov — APPROVED** [review]

File: src/frontends/tensorflow/tests/compilation.cpp [hunk 2]
```diff
@@ -72,12 +72,15 @@ TEST_F(CompileModelsTests, ModelWithSplitConvConcat)
 TEST_F(CompileModelsTests, ModelWithShapeOf) {
     auto model = convert_model("shapeof_slice_abs/shapeof_slice_abs.pbtxt");
     ov::Core core;
+    core.set_property("CPU", ov::hint::inference_precision(ov::element::f32));
```
Comment (ilya-lavrenov) [inline src/frontends/tensorflow/tests/compilation.cpp:75]: should we have a fix on plugin level?

Comment (alvoron) [reply src/frontends/tensorflow/tests/compilation.cpp:75]: CPU plugin on ARM uses f16 inference precision by default for non-convolution networks.
@dmitry-gorokhov do you see a way how to avoid modifying common tests?

File: tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def create_tf_random_uniform_net(self, global_seed, op_seed, x_shape, min_val, m
     @pytest.mark.nightly
     @pytest.mark.precommit
     @pytest.mark.precommit_tf_fe
-    @pytest.mark.xfail(condition=platform.system() == 'Darwin' and platform.machine() == 'arm64',
+    @pytest.mark.xfail(platform.machine() in ["aarch64", "arm64", "ARM64"],
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py:93]: why this change needed?

**Review by rkazants — CHANGES_REQUESTED** [review]
A number of TF FE tests are disabled. Looks strange and incomplete.

File: tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py (hunk 3 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py:93]: Who will address this limitation in the future?

Comment (alvoron) [reply tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py:93]: I have a conversation with the team that supports these tests, there is no way to force f16 inference precision in these tests. It was suggested to skip these tests on arm64 platform.
As soon as inference precision option is added to tests, skip should be removed.

**Comment by alvoron** [comment]
> A number of TF FE tests are disabled. Looks strange and incomplete.

Please let me know if there is a way to specify inference precision in skipped tests. As far as I understand, after having discussions with Jan and Katarzyna there is no such way.

**Review by rkazants — APPROVED** [review]
For layer tests, changes look good

**Review by dmitry-gorokhov — APPROVED** [review]
